scipy/optimize/tests/test_chandrupatla.py

```diff
@@ -655,11 +655,11 @@ def test_convergence(self, xp):
         x1, x2 = bracket
         f0 = xp_minimum(xp.abs(self.f(x1, *args)), xp.abs(self.f(x2, *args)))
         res1 = _chandrupatla_root(self.f, *bracket, **kwargs)
-        xp_assert_less(np.abs(res1.fun), 1e-3*f0)
+        xp_assert_less(xp.abs(res1.fun), 1e-3*f0)
```

Comment:
I was confused about why this was passing with `torch`, but I guess `np.abs` uses `__abs__` or something:
```python3
import numpy as np
import torch
x = torch.asarray([-1, 1.])
np.abs(x)  # tensor([1., 1.])
```

Reply:
> I was confused about why this was passing with `torch` CPU, but ufuncs seem to preserve type.

Only when the input array type implements `__array_wrap__` or some other such method. `torch.Tensor.__array_wrap__` exists, so the return value remains a `torch.Tensor`.